### PR TITLE
microsoft-atp add arrays support

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/44504)
fixes: [link to the issue](https://github.com/demisto/etc/issues/46247)
fixes: [link to the issue](https://github.com/demisto/etc/issues/46460)

## Description
Add array support for the following commands:

- [x] !microsoft-atp-isolate-machine should support also Hostname and IP + Array support for bulk operation.
- [x] !microsoft-atp-unisolate-machine should support also Hostname and IP + Array support for bulk operation.
- [x] microsoft-atp-list-machine-actions-details (support couple of machine IDs / IPs / Hostnames ) can be good to determine what was the last actions on devices
- [x] microsoft-atp-get-file-related-machines : (add array to support couple of hash files - file_hash field) for dedicated file investigation
- [x] microsoft-atp-run-antivirus-scan : run AV scan on multiple devices ( array support on the machine_id field)
- [x] !microsoft-atp-get-machine-details  : Support for multiple machine_ids
- [x] !microsoft-atp-get-machines : Support for multiple IPs / hostnames
- [x] New command: !microsoft-atp-list-machines-by-vulnerability
- [x] New command: !endpoint
- [x] New command: !microsoft-atp-get-file-info
- [x] New command: !microsoft-atp-indicator-batch-update

- [x] Add Uts and edit TPB
- [x] Last design changes.
- [x] New script to support the new indicator command.
- [x] Better docs and UTs for script and integration.

- [x] New command: !microsoft-atp-get-alert-by-id
- [x] Edit command microsoft-atp-list-alerts command will have 2 new args: limit and creation_time. The results will be with the evidence attribute.
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 5.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 1]
```diff
@@ -4517,6 +4522,248 @@ script:
...
+      type: String
+    - contextPath: MicrosoftATP.Machine.IPAddresses.type
+      description: The machine macine IP address type.
+      type: String
+    - contextPath: MicrosoftATP.Machine.AgentVersion
+      description: The machine Agent version.
+      type: String
+  - arguments:
+    - default: false
+      description: 'A JSON object with list of MS defender ATP indicators to update. indicator_batch
```
Comment (bakatzir) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:4743]: https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/ti-indicator?view=o365-worldwide add the link ref please.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md [hunk 2]
```diff
@@ -3776,3 +3998,435 @@ Creates a new indicator.
...
+Ti.ReadWrite
+Ti.ReadWrite.All
+
+##### Limitations
+1. Rate limitations for this API are 30 calls per minute.
+2. There is a limit of 15,000 active indicators per tenant.
+3. Maximum batch size for one API call is 500.
+
+##### Note
+Please read [here](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/ti-indicator?view=o365-worldwide) about a Microsoft Defender for Endpoint indicator resource type
```
Comment (abaumgarten) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:4381]: @bakatzir FYI (related to https://github.com/demisto/content/pull/17315/files#r797346784)

Comment (bakatzir) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:4381]:
```suggestion
Please read [here](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/ti-indicator?view=o365-worldwide) about the Microsoft Defender for Endpoint indicator resource type.
We suggest using the <insert_automation_name>[insert_automation_readme?] to load the XSOAR IOCs to MSDE indicator format.
```

Comment (abaumgarten) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md:4381]: Will do this once I finish developing the automation. thanks!

Comment (abaumgarten) [reply Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/README.md]: Done

**Review by abaumgarten — COMMENTED** [review]
from the CR

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Scripts/TransformIndicatorToMSDefenderIOC/TransformIndicatorToMSDefenderIOC.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+- default: false
+  defaultValue: '0'
+  description: The results offset page. Only change when the number of the results
+    exceed the limit.
+  isArray: false
+  name: offset
+  required: false
+  secret: false
+comment: Transform a XSOAR indicator into a Microsoft Defender for Endpoint IOC. The
+  output (at TransformIndicatorToMSDefenderIOC.JsonOutput) is a json represents the
```
Comment (bakatzir) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Scripts/TransformIndicatorToMSDefenderIOC/TransformIndicatorToMSDefenderIOC.yml:40]:
```suggestion
  output (at TransformIndicatorToMSDefenderIOC.JsonOutput) is a json representation of the
```

**Review by guykeller — APPROVED** [review]
last comments, leaving to your judgement

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_4_0.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+
+#### Integrations
+##### Microsoft Defender for Endpoint
+- Added new commands:
+    - ***microsoft-atp-list-machines-by-vulnerability***.
+    - ***endpoint***.
+    - ***microsoft-atp-get-file-info***.
+    - ***microsoft-atp-indicator-batch-update***.
+    - ***microsoft-atp-get-alert-by-id***.
+- Updated the following commands to get comma separated args:
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_4_0.md:10]:
```suggestion
- Updated the following commands to support a comma separated list as an argument:
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_4_0.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  - In command **microsoft-atp-isolate-machine** now the argument *machine_id* can be a comma separated list.
+  - In command **microsoft-atp-unisolate-machine** now the argument *machine_id* can be a comma separated list.
```
Comment (guykeller) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_4_0.md:12]: Please perform the same change on other suitable locations
```suggestion
  - In the **microsoft-atp-unisolate-machine** command the *machine_id* argument can now be a comma separated list.
```

**Review by guykeller — COMMENTED** [review]
Please fix my notes before merging

**Pull request merged**